picard/mbjson.py

```diff
@@ -383,8 +384,13 @@ def recording_to_metadata(node, m, track=None):
             add_isrcs_to_metadata(value, m)
         elif key == 'video' and value:
             m['~video'] = '1'
+    config = get_config()
     if m['title']:
         m['~recordingtitle'] = m['title']
+    if m['~recordingoriginaldate'] and (
+        config.setting["originaldate_use_recording"] or not m['originaldate']):
+        m['originaldate'] = m['~recordingoriginaldate']
+        m['originalyear'] = m['originaldate'][:4]
     if m.length:
```

Comment:
a. You can still have original date set from recordings (i.e. with different dates across the album - potentially leading to file path issues) even if you have selected Release Group as the source. This is simply BAD!!! and will confuse people.

b. We probably need `~releaseoriginalyear` and `~recordingoriginalyear` if we implement the script updating functionality I have suggested in my global comment.

Reply:
Yes - fallback seems fine for NAT tracks.

We only need `~releaseoriginalyear` and `~recordingoriginalyear` if we implement automatic script renaming - and as you have pointed out there is always a counter example where you will break something if you do this.

Reply:
I changed the behavior and moved setting the originaldate / originalyear only for non-album tracks unconditionally. For releases it will not change the dates to the recording dates unless explicitly enabled.

I also changed the default to stay with the release group date in any case. So if we decide to have an actual option inside the UI I think this would address the most severe concerns. Plus having a warning of some kind maybe.